### PR TITLE
User rate product- add gem ratyrate

### PR DESCRIPTION
**Pull request by longnv5994**

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -70,6 +87,31 @@
...
+    t.string   "dimension"
+    t.datetime "created_at"
+    t.datetime "updated_at"
+  end
+
+  add_index "rating_caches", ["cacheable_id", "cacheable_type"], name: "index_rating_caches_on_cacheable_id_and_cacheable_type"
+
   create_table "users", force: :cascade do |t|
     t.string   "email",                  default: "", null: false
     t.string   "encrypted_password",     default: "", null: false
```
Comment (anhkhang2304) [inline db/schema.rb]: OK

File: app/models/average_cache.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+class AverageCache < ActiveRecord::Base
+  belongs_to :rater, :class_name => "User"
```
Comment (nhunght) [inline app/models/average_cache.rb]: class_name: "User'

File: app/models/average_cache.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
...
+  belongs_to :rateable, :polymorphic => true
```
Comment (nhunght) [inline app/models/average_cache.rb]: polymorphic: true

File: config/routes.rb [hunk 4]
```diff
@@ -1,4 +1,5 @@
 Rails.application.routes.draw do
+  post '/rate' => 'rater#create', :as => 'rate'
```
Comment (nhunght) [inline config/routes.rb]: double quote

File: db/migrate/20160922022645_create_rates.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
+class CreateRates < ActiveRecord::Migration
+
+  def self.up
+      create_table :rates do |t|
```
Comment (nhunght) [inline db/migrate/20160922022645_create_rates.rb]: indent

**Pull request merged**